lib/template-mixins.js

```diff
@@ -85,34 +85,39 @@ module.exports = function (t, fields, options) {
 
     function optionGroup(key) {
         return {
-            key: key,
-            error: this.errors && this.errors[key],
-            options: _.map(fields[key].options, function (label) {
-                var selected = false, value;
-
-                if (typeof label === 'string') {
-                    value = label;
+            'key': key,
+            'error': this.errors && this.errors[key],
+            'options': _.map(fields[key].options, function (obj) {
+                var selected = false, label, value, toggle;
+
+                if (typeof obj === 'string') {
+                    value = obj;
+                    label = obj;
                 } else {
-                    value = label.value;
-                    label = label.label;
+                    value = obj.value;
+                    label = obj.label;
+                    toggle = obj.toggle;
                 }
 
                 if (this.values && this.values[key] !== undefined) {
                     selected = this.values[key] === value;
                 }
+
                 return {
                     label: t(label),
                     value: value,
-                    selected: selected
+                    selected: selected,
+                    toggle: toggle
                 };
             }, this),
-            display: display(key)
+            'display': display(key)
         };
     }
 
     function checkbox(key, opts) {
         opts = opts || {};
         opts.required = opts.required || false;
+        opts.toggle = fields[key] && fields[key].toggle;
```

Comment:
Not sure what exactly you mean. Many times the 'toggle' feature doesn't relate to validation. Often, for example, JavaScript is used to display an info message to a user when they have selected a 'No' choice. Sometimes however it is validation related: for example, an extra input box appearing when you have selected a particular choice (analogous to a dependency with a prerequisite.)

I wonder if it not always being driven by validation changes anything?

However I do not particularly like that we always store information in the `fields.js`. I was being consistent in the code with the fact that there was some other information like this specified there already, for example `display: inline`, but it was also done because there doesn't seem to be a nice way to pass presentation data into a mixin - the only time two bits of information is passed into a mixin is with `{{#input-submit}}` which can take a value and an id. Would it be crazy to do something different like try to parse keys into a mixin as JSON and then falling back to treating them as strings otherwise?